swift/storage.py

```diff
@@ -360,7 +361,8 @@ def _path(self, name):
             path = urlparse.urlsplit(url).path
             msg = ('%s\n%s\n%s' % (method, expires, path)).encode()
             sig = hmac.new(self.temp_url_key, msg, sha1).hexdigest()
-            url = url + '?temp_url_sig=%s&temp_url_expires=%s' % (sig, expires)
+            tmp_path = generate_temp_url(path, expires, sig, method, absolute=True)
```

Comment:
I think we should not pass `sig` here, but `self.temp_url_key`, [according to the docs](http://docs.openstack.org/developer/python-swiftclient/swiftclient.html#swiftclient.utils.generate_temp_url).